examples/ai-image-generator/pages/index.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useEffect } from "react";
+import initMiro from "../initMiro";
+import PromptInput from "../components/PromptInput";
+import Button from "../components/Button";
+
+export const getServerSideProps = async function getServerSideProps({ req }) {
+  const { miro } = initMiro(req);
+
+  // redirect to auth url if user has not authorized the app
+  if (!(await miro.isAuthorized(""))) {
+    return {
+      props: {
+        boards: [],
+        authUrl: miro.getAuthUrl(),
+      },
+    };
+  }
+
+  const api = miro.as("");
+
+  const boards = [];
+
+  for await (const board of api.getAllBoards()) {
+    boards.push(board.name || "");
+  }
+
+  return {
+    props: {
+      boards,
+    },
+  };
+};
+
+export default function Main({ boards, authUrl }) {
+  useEffect(() => {
+    if (new URLSearchParams(window.location.search).has("panel")) return;
+
+    window.miro.board.ui.on("icon:click", async () => {
+      window.miro.board.ui.openPanel({
+        url: `/?panel=1`,
+      });
+    });
+  }, []);
+
+  // Register the drop event handler once.
+  useEffect(() => {
+    window.miro.board.ui.on("drop", drop);
+    console.log("drop event is running");
+  }, []);
+
+  //drag and drop logic
+  const drop = async (e) => {
+    const { x, y, target } = e;
+
+    if (target instanceof HTMLImageElement) {
+      console.log("about to create image");
+      const image = await window.miro.board.createImage({
+        x,
+        y,
+        url: target.src,
+      });
+      await window.miro.board.viewport.zoomTo(image);
+    }
+  };
+
+  const [inputValue, setInputValue] = useState("");
+
+  //handles the response being returned from backend (URL of image)
+  const [response, setResponse] = useState("");
+
+  //handles the prompt input being typed in
+  const handleInputChange = (newValue) => {
+    setInputValue(newValue);
+  };
+
+  // Shows spinner while API calls are in progress
+  function showSpinner() {
+    document.querySelector(".spinner").classList.add("show");
+  }
+
+  // Removes spinner when API calls are finished and data is returned
+  function removeSpinner() {
+    document.querySelector(".spinner").classList.remove("show");
+  }
+
+  const handleButtonClick = async () => {
```

Comment:
would be nice to see the same code style for all functions in the example. Currently, some of them are declared as a function (`function removeSpinner() {`), but some as a variable: `const handleButtonClick = async () => {`

Reply:
great point - thank you for this!

Reply:
Ah - sorry still need to update this one, just one second @kirillsud

Reply:
ok should be ready now